packages/react-html/src/components/Meta.tsx

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import {Effect} from '@shopify/react-effect';
+import {Consumer} from '../context';
+import {EFFECT_ID} from '../utilities';
+import Manager from '../manager';
+
+type Props = React.HTMLProps<HTMLMetaElement>;
+
+export const EXTRACT_ID = Symbol('html');
+
+export default class Meta extends React.PureComponent<Props> {
+  private removeMeta?: ReturnType<InstanceType<typeof Manager>['addMeta']>;
+
+  componentWillUnmount() {
+    if (this.removeMeta) {
+      this.removeMeta();
+    }
+  }
+
+  render() {
+    return (
+      <Consumer>
+        {manager => (
+          <Effect
+            key={JSON.stringify(this.props)}
```

Comment:
This is a bit lazy, but it basically just makes it so we don't have to care about the props changing case (since the component will be unmounted and remounted, re-executing `perform` as a result.